File: services/content-service/src/main/java/com/novix/content_service/service/CategoryService.java
package com.novix.content_service.service;

import com.novix.content_service.dto.request.CreateCategoryRequest;
import com.novix.content_service.dto.request.UpdateCategoryRequest;
import com.novix.content_service.dto.response.PageResponse;
import com.novix.content_service.entity.Category;

public interface CategoryService {
    Category createCategory(CreateCategoryRequest request);

    Category getCategoryById(Long id);

    Category getCategoryBySlug(String slug);

    PageResponse<Category> getAllCategories(int page, int size);

    Category updateCategory(UpdateCategoryRequest request);

    void deleteCategory(Long id);
}

File: frontend/admin/src/services/category.service.ts
import api from '@/lib/api';
import { Category, PageResponse } from '@/types';

export const categoryService = {
  async getCategories(page = 0, size = 20): Promise<PageResponse<Category>> {
    const response = await api.get<PageResponse<Category>>(`/api/v1/categories?page=${page}&size=${size}`);
    return response.data;
  },

  async createCategory(name: string, description: string, slug: string): Promise<Category> {
    const response = await api.post<Category>(
      `/api/v1/categories?name=${encodeURIComponent(name)}` +
        `&description=${encodeURIComponent(description)}` +
        `&slug=${encodeURIComponent(slug)}`,
    );
    return response.data;
  },

  async deleteCategory(id: number): Promise<void> {
    await api.delete(`/api/v1/categories/${id}`);
  },
};
